src/Microsoft.DotNet.Build.Tasks/FindBestConfigurations.cs

```diff
@@ -42,7 +40,18 @@ public override bool Execute()
 
                 var compatibleConfigurations = ConfigurationFactory.GetCompatibleConfigurations(buildConfiguration, DoNotAllowCompatibleValues);
 
-                var bestConfiguration = compatibleConfigurations.FirstOrDefault(c => supportedProjectConfigurations.Contains(c));
+                bool isPlaceHolderConfiguration = false;
+                var bestConfiguration = compatibleConfigurations.FirstOrDefault(c =>
+                {
+                    var supportedConfiguration = supportedProjectConfigurations.FirstOrDefault(sc => Configuration.CompatibleComparer.Equals(sc, c));
```

Comment:
Searching linearly might end up being a perf issue but for now I don't think it it really matters.

Reply:
I changed to use Dictionary instead to avoid linear search and avoid future issues.